src/Client/ConnectionPoolWithFailover.h

```diff
@@ -31,6 +31,55 @@ enum class PoolMode
     GET_ALL
 };
 
+/// Class for establishing connection with replica without blocking.
+class TryGetConnection
```

Comment:
Class name is not very obvious. We don't get connection, we are trying to establish it using different stages indeed.
Better use nouns for class names.